### PR TITLE
Add app token QR codes for mobile apps

### PR DESCRIPTION
**Pull request by ChristophWurst**
This adds the option to generate a QR code in order to easily set up a mobile app with a new app token. App tokens are allowed on mobile as well, but in some cases it's easier to do this flow, e.g. when you're using U2F as your only 2FA method. In that case you would have to type the newly created app token into your phone.

Fixes https://github.com/nextcloud/server/issues/429.

Todo
- [x] Add subtle button to show/generate a QR code
- [x] Render a URL that actually works with the mobile app

cc @tobiasKaminsky @nextcloud/android @nextcloud/ios 

---

From https://github.com/nextcloud/server/issues/429#issuecomment-442495238

> waiting desperately for integration on server by @ChristophWurst

The time has come :rocket:

### CODE REVIEW

**Comment by ChristophWurst** [comment]
# Screen 1
bildschirmfoto von 2019-03-01 09-13-14

# Screen 2
bildschirmfoto von 2019-03-01 09-13-25

# Screen 3
bildschirmfoto von 2019-03-01 09-13-33

**Comment by ChristophWurst** [comment]
Found out the QR reader feature is already included in the latest Android app releases: https://github.com/nextcloud/server/issues/429.

This is ready to review. I could successfully set up an account on Android with my local dev instance :rocket:

**Comment by ChristophWurst** [comment]
@marinofaggiana are the `nc://login` URLs supported in our iOS app as well? Is there a QR code reader?

**Review by rullzer — APPROVED** [review]
Works like a charm!

**Comment by ChristophWurst** [comment]
Failing files app acceptance test is unrelated.

File: settings/src/components/AuthTokenSetupDialogue.vue [hunk 1]
```diff
@@ -63,14 +63,31 @@
 				{{ t('settings', 'Done') }}
 			</button>
 		</div>
+		<div class="app-password-row"
+			 v-if="!showQR">
```
Comment (skjnldsv) [inline settings/src/components/AuthTokenSetupDialogue.vue]: v-if, v-else?

**Review by skjnldsv — CHANGES_REQUESTED** [review]
One little change but fine otherwise!!

File: settings/src/components/AuthTokenSetupDialogue.vue (hunk 1 above)
Comment (ChristophWurst) [reply settings/src/components/AuthTokenSetupDialogue.vue]: Fixed!

**Review by skjnldsv — APPROVED** [review]

**Comment by ChristophWurst** [comment]
CI failure is unrelated

**Comment by marinofaggiana** [comment]
Available on iOS.

**Pull request merged**